### PR TITLE
feat: Implement frontend function for pytorch linalg diagonal

### LINKED ISSUE #22531: diagonal

### PR DESCRIPTION
**Pull request by rajendrabaskota**
Close #22531

### CODE REVIEW

**Comment by rajendrabaskota** [comment]
Hi @satakshigarg, I had a closer look into the torch.linalg.diagonal function and it is defined as an alias of torch.diagonal which is defined in the miscellaneous_ops.py file of the torch frontend. There is also a test script defined for it in the test_miscellaneous_ops.py file but there seems to be no test script for the frontend diagonal function of linalg.py which this issue relates to. Should I add the test for linalg.diagonal or is it not necessary since the linalg.diagonal is just an alias of the diagonal function?

**Comment by satakshigarg** [comment]
Thanks @rajendrabaskota for your contributions 😃
Apologies for the delay in response. The changes look good to me, except for few failing tests. It seems that unsupported data types by PyTorch are being produced somewhere. Could you please resolve them, then the PR would be good to merge.
Thanks !!

Keep contributing !!
Screenshot 2023-09-03 at 12 53 53 AM

**Comment by KevinUli** [comment]
Hi @rajendrabaskota , thanks for the PR. Could you include `offset`, `dim1`, and `dim2` in your tests. Thanks! Feel free to re-request a review after the change.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py
Comment (KevinUli) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:1]: You seem to have added a helper function for `dims` and `offset` but maybe forgot to add it into the unit test function.

Comment (KevinUli) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:1]: The test seems to be failing. Please fix the failing tests and feel free to re-request a review afterwards.

**Comment by Ishticode** [comment]
Closing this due to lack of activity after changes were requested. Please feel free to reopen if you would like to continue. Thank you very much :)

**Pull request closed without merging**